packages/react-tabs/src/components/Pivot/Pivot.base.tsx

```diff
@@ -66,13 +65,25 @@ const isPivotItem = (item: React.ReactNode): item is PivotItem => {
 
 export const PivotBase: React.FunctionComponent<IPivotProps> = React.forwardRef<HTMLDivElement, IPivotProps>(
   (props, ref) => {
-    const { componentRef, theme, linkSize, linkFormat, overflowBehavior } = props;
-    const pivotId: string = useId('Pivot');
-    let linkCollection = getLinkItems(props, pivotId);
     const focusZoneRef = React.useRef<IFocusZone>(null);
-    const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(props, divProperties);
+    const overflowMenuButtonComponentRef = React.useRef<IButton>(null);
+    const pivotId: string = useId('Pivot');
+
     const [selectedKey, setSelectedKey] = useControllableValue(props.selectedKey, props.defaultSelectedKey);
+
+    const { componentRef, theme, linkSize, linkFormat, overflowBehavior } = props;
+    // The overflow menu starts empty and items[] is updated as the overflow items change
+    const overflowMenuProps: IContextualMenuProps = {
+      items: [],
+      doNotLayer: true,
+      alignTargetEdge: true,
+      directionalHint: DirectionalHint.bottomRightEdge,
+    };
+
     let classNames: { [key in keyof IPivotStyles]: string };
+    const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(props, divProperties);
+
+    let linkCollection = getLinkItems(props, pivotId);
```

Comment:
(nit) out of curiosity, why are you moving around so many lines of code in this function? It's fine to clean things up if the code is messy, but this doesn't necessarily seem better than it was before. (E.g. previously, overflowMenuProps was declared closer to where it's used, which makes more sense to me). It's not necessarily a problem, but it seems unnecessary and makes the diff harder to read.

Reply:
I reverted the overflowMenuProps move and the diffing should be improved. The intention is to help keep consistency in Pivot’s useRef and useState/useControllableValue placement to follow patterns in other functional components.